File: server/src/main/java/com/example/server/backend/UsPinterest/dto/ProfileResponse.java
package com.example.diplomwork.backend.UsPinterest.dto;

import java.util.List;

public class ProfileResponse {
    private String username;
    private String email;
    private List<PinResponse> pins;

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public List<PinResponse> getPins() {
        return pins;
    }

    public void setPins(List<PinResponse> pins) {
        this.pins = pins;
    }
}

File: app/src/main/java/com/example/diplomwork/model/ProfileResponse.kt
package com.example.diplomwork.model

data class ProfileResponse(
    val id: Long,
    val username: String,
    val email: String,
    val profileImageUrl: String? = null,
    val pins: List<PictureResponse>,
    val pinsCount: Int = 0,
    val followersCount: Int = 0,
    val followingCount: Int = 0,
    val bio: String? = null
)